images/Dockerfile.ubuntu-20-lts-arm-rpi

```diff
@@ -0,0 +1,79 @@
+ARG BASE_IMAGE=ubuntu:20.04
+ARG MODEL=generic
+
+FROM $BASE_IMAGE as base
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update \
+    && apt-get install -y software-properties-common \
+    && add-apt-repository ppa:oibaf/test \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+FROM base AS generic
+RUN apt-get update && apt-get install -y linux-base \
+    linux-image-generic-hwe-22.04 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+FROM base AS rpi64
+# Normally we install also linux-modules-extra-raspi but this is not available for 20.04, not sure what's the impact to
+# remove it.
+RUN apt-get update && apt-get install -y linux-firmware-raspi2 \
+    linux-raspi \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+FROM ${MODEL} AS all
```

Comment:
I...dont get this.

So if we call this with the generic model it will run:
 - FROM $BASE_IMAGE as base
 - FROM base AS generic
 - FROM ${MODEL} AS all

but all its still installing the `linux-raspi` kernel no? dont we want the rpi to install those packages only? Wont we end up with a generic image that has both kernels inside and the possibility of linking to it?

Reply:
Yeah I didn't find it very intuitive, my understanding is that `all` is the requested target (because it's the last one?) and depending on it's base it will only build generic or rpi64. And it properly ignores all the targets it doesn't need to build

see: https://docs.docker.com/build/building/multi-stage/#differences-between-legacy-builder-and-buildkit